backend/canisters/identity/impl/src/model/identity_map.rs

```diff
@@ -18,4 +20,37 @@ impl IdentityMap {
     pub fn get_by_principal(&self, principal: &Principal) -> Option<&Identity> {
         self.identities_by_principal.get(principal)
     }
+
+    pub fn try_register_phone_number(
+        &mut self,
+        principal: Principal,
+        phone_number: PhoneNumber,
+        now: TimestampMillis,
+    ) -> Option<FieldId> {
+        if self.registered_phone_numbers.contains(&phone_number) {
+            return None;
+        }
+
+        let id = FieldId::new();
+        let field = VerifiableField::<PhoneNumber> {
+            id,
+            status: VerificationCodeStatus::Pending,
+            added: now,
+            value: phone_number.clone(),
+        };
+
+        match self.identities_by_principal.entry(principal) {
+            Occupied(mut e) => {
+                let identity = e.get_mut();
```

Comment:
You should use `into_mut` here.
From my brief reading about the difference a while ago the general consensus was that you should use `into_mut` by default and only use `get_mut` if you need to reference the `OccupiedEntry` again later on.
`into_mut` will consume the `OccupiedEntry` so you don't need to mark it as `mut`.